api/permissions.py

```diff
@@ -15,10 +15,16 @@ def has_permission(self, request, view):
         return is_authenticated and not isinstance(request.successful_authenticator, OAuth2Authentication)
 
 
-class IsAuthenticatedOrTokenHasResourceScope(TokenHasResourceScope):
+class IsAuthenticatedOrTokenHasResourceScope(permissions.BasePermission):
     def has_permission(self, request, view):
         is_authenticated = IsAuthenticated().has_permission(request, view)
-        return is_authenticated or super().has_permission(request, view)
+
+        oauth2authenticated = False
+        if is_authenticated:
+            oauth2authenticated = isinstance(request.successful_authenticator, OAuth2Authentication)
+
+        token_has_scope = TokenHasResourceScope()
+        return (is_authenticated and not oauth2authenticated) or token_has_scope.has_permission(request, view)
```

Comment:
I got a bit stuck on understanding this line, is the following the same logic?
`return token_has_scope... if oauth2authenticated else is_authenticated`
I find it a bit clearer to translate to English "Use the token permission for oauth2, otherwise use normal"

Reply:
Adding a variable for the first half of the condition over here : https://github.com/betagouv/ma-cantine/pull/2442/commits/d5074b6ad66ff31aa8447ac9a58cd02b3e972dd7